libraries/ms-common/test/test_critical_section.c

```diff
@@ -23,3 +28,13 @@ void test_interrupt_disable_enable(void) {
   critical_section_end(disabled);
   TEST_ASSERT_TRUE(critical_section_start());
 }
+
+void test_critical_section_cleanup(void) {
+  // Starts critical section but does not explicitly end it
+  prv_cleanup_section();
```

Comment:
Instead of a helper function why not use an unnamed/anonymous scope block? Just helps keep things neater in the test.

i.e.
```c
{
  CRITICAL_SECTION_AUTOEND;
  TEST_ASSERT_TRUE(_disabled);
}
```

Reply:
I tried that and it doesn't seem to work. That was my initial test :)